src/StreamDeckLib/ConnectionManager.cs

```diff
@@ -1,300 +1,360 @@
-					profile = profileName
-				}
-			};
-
-			await _Proxy.SendStreamDeckEvent(args);
-
-		}
-
-		public async Task OpenUrlAsync(string context, string url)
-		{
-			var args = new OpenUrlArgs()
-			{
-				context = context,
-				payload = new OpenUrlArgs.Payload()
-				{
-					url = url
-				}
-			};
-			await _Proxy.SendStreamDeckEvent(args);
-		}
-
-		#endregion
-
-		#region IDisposable Support
-
-		private bool disposedValue = false; // To detect redundant calls
-		private BaseStreamDeckPlugin _Plugin;
-		private static ILoggerFactory _LoggerFactory;
-		private static ILogger _Logger;
-
-		void Dispose(bool disposing)
-		{
-			if (!disposedValue)
-			{
-				if (disposing)
-				{
-					_Proxy.Dispose();
-				}
-
-				disposedValue = true;
-			}
-		}
-
-		// This code added to correctly implement the disposable pattern.
-		public void Dispose()
-		{
-			// Do not change this code. Put cleanup code in Dispose(bool disposing) above.
-			Dispose(true);
-			// TODO: uncomment the following line if the finalizer is overridden above.
-			// GC.SuppressFinalize(this);
-		}
-		#endregion
-	}
-}
+﻿using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Newtonsoft.Json;
+using StreamDeckLib.Messages;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics;
+using System.IO;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace StreamDeckLib
+{
+    /// <summary>
+    /// This class manages the connection to the StreamDeck hardware
+    /// </summary>
+    public partial class ConnectionManager : IDisposable
+    {
+        private int _Port;
+        private string _Uuid;
+        private string _RegisterEvent;
+		private IStreamDeckProxy _Proxy;
+        private ConnectionManager() { }
+        public Messages.Info Info { get; private set; }
+        public static ConnectionManager Initialize(string[] commandLineArgs,
+				ILoggerFactory loggerFactory = null, IStreamDeckProxy streamDeckProxy = null)
+        {
+			var waitForDebugger = false;
+            using (var app = new CommandLineApplication())
+            {
+                app.HelpOption();
+
+                var optionPort = app.Option<int>("-port|--port <PORT>", "The port the Elgato StreamDeck software is listening on", CommandOptionType.SingleValue);
+                var optionPluginUUID = app.Option("-pluginUUID <UUID>", "The UUID that the Elgato StreamDeck software knows this plugin as.", CommandOptionType.SingleValue);
+                var optionRegisterEvent = app.Option("-registerEvent <REGEVENT>", "The registration event", CommandOptionType.SingleValue);
+                var optionInfo = app.Option("-info <INFO>", "Some information", CommandOptionType.SingleValue);
+
+#if DEBUG
+				var optionDebug = app.Option<bool>("-d|--debug", "Wait for a debugger to attach before execution", CommandOptionType.NoValue);

```

Comment:
good idea!

Reply:
Thanks. Wanted to make sure we kept this **strictly** within the development process.